src/Symfony/Component/PropertyInfo/PhpStan/NameScopeFactory.php

```diff
@@ -22,21 +22,36 @@ final class NameScopeFactory
 {
     public function create(string $fullClassName): NameScope
     {
+        $reflection = new \ReflectionClass($fullClassName);
         $path = explode('\\', $fullClassName);
         $className = array_pop($path);
-        [$namespace, $uses] = $this->extractFromFullClassName($fullClassName);
+        [$namespace, $uses] = $this->extractFromFullClassName($reflection);
 
-        foreach (class_uses($fullClassName) as $traitFullClassName) {
-            [, $traitUses] = $this->extractFromFullClassName($traitFullClassName);
-            $uses = array_merge($uses, $traitUses);
-        }
+        $uses = array_merge($uses, $this->collectUses($reflection));
 
         return new NameScope($className, $namespace, $uses);
     }
 
-    private function extractFromFullClassName(string $fullClassName): array
+    private function collectUses(\ReflectionClass $reflection): array
+    {
+        [, $classUses] = $this->extractFromFullClassName($reflection);
+        $uses = [$classUses];
+
+        foreach (class_uses($reflection->name) as $traitFullClassName) {
+            [, $traitUses] = $this->extractFromFullClassName(new \ReflectionClass($traitFullClassName));
+            $uses[] = $traitUses;
+        }
+
+        $parentClass = $reflection->getParentClass();
+        if (false !== $parentClass) {
```

Comment:
should be merged with the previous line:
`if (false !== $parentClass = $reflection->getParentClass()) {`